### DISCUSSION TITLE
https://1kevinson.com/create-a-base-entity-with-jpa/

### DISCUSSION THREAD

**Discussion by giscus** [1kevinson/BLOG-TUTORIALS#15] [Announcements]
# https://1kevinson.com/create-a-base-entity-with-jpa/

In this tutorial, you will learn how to define a basic template for all your JPA/Hibernate entity classes.

https://1kevinson.com/create-a-base-entity-with-jpa/

**Comment by nhducseuit** [comment]
Hi Kevin,

Firstly thanks for your tutorial on creating base entity classes.
I wonder if there is a reason for not having constructor or any @ArgConstructors... annotation from Lombok in your sample code?

Thanks again.

**Reply by 1kevinson** [reply]
Hi mate!

Glad the article helped you.
For your question, if no other constructor is defined, a default no-arg constructor will be provided automatically at compile time of your application. That's why I have not put it in this article.

However, if you want to define a parameterized constructor `@ArgConstructors`, you must also provide a default constructor explicitly.

👀 If you would like to dive deeper on this topic, I recommend you this article on Baeldung : 
[Need for Default Constructor in JPA Entities](https://www.baeldung.com/jpa-no-argument-constructor-entity-class#jpa-specification-constructors-and-reflection)
